docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1748 @@
+
+.. raw:: html
+
+    </details>
+
+
+.. raw:: html
+
+    <details>
+    <summary><a><font size="+1">Example 2: Evolution with time-dependent Hamiltonian</font></a></summary>
+    <br>
+
+**Opflow**
+
+.. code-block:: python
+
+    >>> from qiskit.opflow import Trotter, PauliTrotterEvolution, PauliSumOp
+    >>> from qiskit.circuit import Parameter
+
+    >>> time = Parameter('t')
+    >>> hamiltonian = PauliSumOp.from_list([('X', 1), ('Y',1)])
+    >>> evolution = PauliTrotterEvolution(trotter_mode=Trotter(), reps=1)
+    >>> evol_result = evolution.convert((time * hamiltonian).exp_i())
+    >>> evolved_state = evol_result.to_circuit()
+
+    >>> print(evolved_state)
+       ┌─────────────────────────┐
+    q: ┤ exp(-it (X + Y))(1.0*t) ├
+       └─────────────────────────┘
+
+**Alternative**
+
+.. code-block:: python
+
+    >>> from qiskit.quantum_info import SparsePauliOp
+    >>> from qiskit.synthesis import LieTrotter
+    >>> from qiskit.circuit.library import PauliEvolutionGate
+    >>> from qiskit import QuantumCircuit
+    >>> from qiskit.circuit import Parameter
+
+    >>> time = Parameter('t')
+    >>> hamiltonian = SparsePauliOp.from_list([('X', 1), ('Y',1)])
+    >>> evol_gate = PauliEvolutionGate(hamiltonian, time=time, synthesis=LieTrotter())
+    >>> evolved_state = QuantumCircuit(1)
+    >>> evolved_state.append(evol_gate, [0])
+
+    >>> print(evolved_state)
+       ┌─────────────────────┐
+    q: ┤ exp(-it (X + Y))(t) ├
+       └─────────────────────┘
+
+.. raw:: html
+
+    </details>
+
+
+.. raw:: html
+
+    <details>
+    <summary><a><font size="+1">Example 3: Matrix evolution</font></a></summary>
+    <br>
+
+**Opflow**
+
+.. code-block:: python
+
+    >>> from qiskit.opflow import MatrixEvolution, MatrixOp
+
+    >>> hamiltonian = MatrixOp([[0, 1], [1, 0]])
+    >>> evolution = MatrixEvolution()
+    >>> evol_result = evolution.convert(hamiltonian.exp_i())
+    >>> evolved_state = evol_result.to_circuit()
+
+    >>> print(evolved_state.decompose().decompose())
+       ┌────────────────┐
+    q: ┤ U3(2,-π/2,π/2) ├
+       └────────────────┘
+
+**Alternative**
+
+.. code-block:: python
+
+    >>> from qiskit.quantum_info import SparsePauliOp
+    >>> from qiskit.extensions import HamiltonianGate
+    >>> from qiskit import QuantumCircuit
+
+    >>> evol_gate = HamiltonianGate([[0, 1], [1, 0]], 1)
+    >>> evolved_state = QuantumCircuit(1)
+    >>> evolved_state.append(evol_gate, [0])
+
+    >>> print(evolved_state.decompose().decompose())
+       ┌────────────────┐
+    q: ┤ U3(2,-π/2,π/2) ├
+       └────────────────┘
+
+.. raw:: html
+
+    </details>
+
+Expectations
+============
+*Back to* `Contents`_
+
+Expectations are converters which enable the computation of the expectation value of an observable with respect to some state function.
+This functionality can now be found in the Estimator primitive.
+
+Algorithm-Agnostic Expectations
+-------------------------------
+*Back to* `Contents`_
+
```

Comment:
Maybe here is a good place to remind people that there are different implementations of primitives as noted `attention` box in the line 82. Maybe link to that.